documents/invoice/model.go

```diff
@@ -319,13 +317,18 @@ func (i *Invoice) UnpackCoreDocument(coreDoc *coredocumentpb.CoreDocument) error
 	}
 
 	i.loadFromP2PProtobuf(invoiceData)
-	invoiceSalts := &invoicepb.InvoiceDataSalts{}
-	err = proto.Unmarshal(coreDoc.EmbeddedDataSalts.Value, invoiceSalts)
-	if err != nil {
-		return err
-	}
 
-	i.InvoiceSalts = invoiceSalts
+	if coreDoc.EmbeddedDataSalts == nil {
```

Comment:
we are changing the policy here to generate salts if empty vs failing. This could cause silent failures when documents get out of sync. Agree that this problem should have been caught way before initially saving it, but I believe this approach will potentially make data inconsistent across the network. What is the main reason for this change?

Reply:
I will check, as agreed